mirrord/sip/src/lib.rs

```diff
@@ -1,3 +1,31 @@
+//! This crate concerns MacOS' SIP (System Integrity Protection) mechanism and how to sidestep it.
+//!
+//! ### Why?
+//!
+//! SIP makes it such that we are unable to load the `mirrord-layer` lib dynamically into the user's
+//! process onto MacOS machines. This is ostensibly a security measure to prevent the malicious or
+//! unintended modification of processes, but mirrord's dynamic library loading is neither. To get
+//! around this, we patch the user's process by creating a new unprotected version and signing it
+//! ourselves. Then we can load the layer dynamically into our unprotected copy and continue as
+//! usual. If the process is a script, the shebang must be pointed to the patched version of the
+//! interpreter.
```

Comment:
Why do we need to load mirrord-layer lib? What does that even mean? I think this would be a good candidate point to link this doc to the place where we put an explanation of what has to happen (probably the crate doc for mirrord-layer itself). So this is not feedback saying that you should change this text per say, instead it's more of a future note.

I would also add some link that explains macos SIP around here, so you can look at more details. Think you can shorten the SIP explanation to just say that it blocks modification of processes. The focus here is how it impacts mirrord, and what the crate is doing to get us around this fact (which you are doing in this paragraph).